src/sql/workbench/contrib/notebook/browser/cellViews/linkHandler.directive.ts

```diff
@@ -65,8 +65,14 @@ export class LinkHandlerDirective {
 				this.notebookService.navigateTo(this.notebookUri, uri.fragment);
 			} else {
 				if (uri.scheme === 'file') {
-					let exists = await this.fileService.exists(uri);
-					if (!exists) {
+					// this will check to see if both cases where file contains %20 (encoded) and file contains spaces (decoded) exists
+					let encodedFileExists = await this.fileService.exists(uri);
+					let decodedFileExists = await this.fileService.exists(URI.parse(content));
+					// If the encoded file is not found then we need to check if decoded path is found and set the uri to open that file
+					if (!encodedFileExists && decodedFileExists) {
```

Comment:
A path with a space and a path with a %20 aren't equivalent though - someone could have both. Or just one of them. This logic doesn't seem to be handling that correctly though - did you test out scenarios with a mix of those both? We don't want to have the links opening the wrong file

Reply:
Yes, there would be an issue regarding this scenario as the href that we get initially from the target htmlElement onclick would both be encoded and will have to find a way to differentiate which path contains the filename via the target.